lib/kitchen/provisioner/chef_zero_shopify.rb

```diff
@@ -1,35 +1,71 @@
 require 'kitchen/provisioner/chef_zero'
 require 'chef-config/config'
 require 'chef/role'
+require 'chef/data_bag_item'
+require 'chef/encrypted_data_bag_item'
+require 'chef/encrypted_data_bag_item/check_encrypted'
 
 module Kitchen
   module Provisioner
-
     # We'll sneak some code in before the default chef zero provisioner runs
     # This will allow us to convert our roles to JSON, and flatten at once
     class ChefZeroShopify < ChefZero
       def create_sandbox
-        tmpdir = Dir.mktmpdir('chef-flattened-roles')
+        tmpdir = Dir.mktmpdir('chef-shopify-inject')
 
         at_exit do
           FileUtils.rm_rf(tmpdir)
         end
 
+        flatten_roles(config, tmpdir) unless config[:roles_path].nil?
+        decrypt_data_bags(config, tmpdir) unless config[:data_bags_path].nil?
+        super
+      rescue => e
+        puts e.message
+        puts e.backtrace
+        raise 'Failed monkey patch the shopify provisioner!'
```

Comment:
We aren't _really_ monkeypatching here... we're extending the functionality of the provisioner.  /nitpick